security/fma-l1-pectra.md

```diff
@@ -0,0 +1,85 @@
+# Upgrade 12 (L1 Pectra Defense): Failure Modes and Recovery Path Analysis
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Generic Items](#generic-items)
+- [Audit Requirements](#audit-requirements)
+- [Action Items](#action-items)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+|                     |                      |
+| ------------------- | -------------------- |
+| Author              | George Knee          |
+| Created at          | 5th February 2025    |
+| Needs Approval From |                      |
+| Other Reviewers     |                      |
+| Status              | Draft                |
+
+## Introduction
+
+This document is intended to be shared in a public space for reviews and visibility purposes. It covers Upgrade 12, which involves the following changes:
+
+- (Consensus Layer) Ability to parse and validate blocks and receipts with EIP-7702 transactions, as well as blocks with a non-nil EIP-7685 requestsHash field. 
+- (Smart Contracts) For chains using Fault Proofs, L1 contracts updates which reference an updated absolute prestate hash
+
+Each change has its own section below with a list of Failure Modes.
+
+Below are references for this project:
+
+- [Github tracker](https://github.com/orgs/ethereum-optimism/projects/117/views/9)
+- [Specs clarification (Logs)](https://specs.optimism.io/protocol/derivation.html#on-future-proof-transaction-log-derivation)
+- [Specs clarification (L1 Retrieval)](https://specs.optimism.io/protocol/derivation.html#l1-retrieval)
+
+## Failure Modes and Recovery Paths
+
+### FM1: Incidental breaking changes from upstream geth
+
+- **Description:** Incidental changes from merging upstream geth cause unexpected problems. Example: database schema change can break op-geth nodes on rollout.
+- **Risk Assessment:** Medium severity, Medium Likelihood
+- **Mitigations:**:
+  1. Our usual staggered rollout and baking processes should catch most problems early.
+  2. Prepare appropriate runbook(s) detailing how nodes may need to be carefully migrated.
+- **Detection:** Alerts from our own infrastructure and/or that of partners'.
+- **Recovery Path(s)**: Follow the above runbook.
+
+### FM2: Consensus bug specific to Pectra upgrade on L1 (client software)
```

Comment:
This could be worded more clearly. Essentially that we forgot to take account of some feature of Pectra activating on L1.